src/crypto.rs

```diff
@@ -119,27 +127,35 @@ where
     }
 
     fn unseal(&self, secret_box: Self::SecretBox) -> Result<SecStr, Self::Error> {
-        ensure_initialised();
-
         let passphrase = self
             .pinentry
             .get_passphrase()
             .map_err(SecretBoxError::Pinentry)?;
 
-        secretbox::open(
-            &secret_box.sealed,
-            &secret_box.nonce,
-            &derive_key(&secret_box.salt, &passphrase),
-        )
-        .map_err(|()| SecretBoxError::InvalidKey)
-        .map(SecStr::new)
+        let derived = derive_key(&secret_box.salt, &passphrase);
+        let key = chacha20poly1305::Key::from_slice(&derived[..]);
+        let cipher = chacha20poly1305::ChaCha20Poly1305::new(key);
+
+        cipher
+            .decrypt(&secret_box.nonce, secret_box.sealed.as_slice())
+            .map_err(|_| SecretBoxError::InvalidKey)
+            .map(SecStr::new)
     }
 }
 
-fn derive_key(salt: &pwhash::Salt, passphrase: &SecUtf8) -> secretbox::Key {
-    let mut k = secretbox::Key([0; secretbox::KEYBYTES]);
-    let secretbox::Key(ref mut kb) = k;
-    pwhash::derive_key_interactive(kb, passphrase.unsecure().as_bytes(), salt)
-        .expect("Key derivation failed"); // OOM
-    k
+fn derive_key(salt: &Salt, passphrase: &SecUtf8) -> [u8; 32] {
```

Comment:
Could we take the `ScryptParams` as a Param and use `recommended` in the production call and the test params in the test call? Or would that mean the parameter bubbles all the way up?

Reply:
Yeah it'll bubble up I'm afraid :/

At least this way it's tucked away and harder to misuse

Reply:
Fair enough, appreciate the comment then :)